packages/editor/src/components/post-text-editor/index.js

```diff
@@ -47,14 +57,32 @@ export default function PostTextEditor() {
 		setIsDirty( true );
 	};
 
+	/**
+	 * Ensures no unwanted HTML ends up in the list of parsed blocks.
+	 */
+	const cleanBlocks = useCallback( function ( blocks ) {
+		return blocks.map( ( block ) => ( {
+			...block,
+			originalContent: block.originalContent
+				? sanitize( block.originalContent )
+				: undefined,
+			attributes: {
+				...block.attributes,
+				content: block.attributes?.content
```

Comment:
Why `content`? It's merely a loose convention that blocks name their "main" attribute `content`. A counterexample is `core/quote`, which has two RichText-based attributes: `value` and `citation`. It might make more sense to target attributes with `source: 'html'`.